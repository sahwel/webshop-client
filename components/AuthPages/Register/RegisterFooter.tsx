import Link from 'next/link';
import { FunctionComponent, useCallback, MouseEvent } from 'react';
import Button from '../../common/Form/Button/Button';
import { Section } from './services/definitions';

interface RegisterFooterProps {
  handleSectionChange: (toFoward: boolean) => void;
  section: Section;
}

const RegisterFooter: FunctionComponent<RegisterFooterProps> = ({
  handleSectionChange,
  section,
}) => {
  const sectionChangeToFoward = useCallback(
    (event: MouseEvent<HTMLElement>) => {
      section < Section.thirdSection && event.preventDefault();
      handleSectionChange(true);
    },
    []
  );

  const sectionChangeToBackward = useCallback(() => {
    handleSectionChange(false);
  }, []);

  return (
    <div>
      <Button
        children="Next"
        type={section >= Section.secondSection ? 'submit' : 'button'}
        onClick={sectionChangeToFoward}
      />
      {section === Section.firstSection && (
        <Link href="/login" passHref>
          <a className="text-center block text-sm mt-2 cursor-pointer">Back</a>
        </Link>
      )}
      {section >= Section.secondSection && (
        <p
          onClick={sectionChangeToBackward}
          className="text-center text-sm mt-2 cursor-pointer"
        >
          Back
        </p>
      )}
    </div>
  );
};

export default RegisterFooter;
