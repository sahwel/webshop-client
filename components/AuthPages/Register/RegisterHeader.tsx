import React from 'react';

const RegisterHeader = () => {
  return (
    <div className="select-none">
      <h1 className="text-[300%]">Let’s sign up!</h1>
      <h3>We are happy you’re</h3>
      <h3>joining us!</h3>
    </div>
  );
};

export default RegisterHeader;
