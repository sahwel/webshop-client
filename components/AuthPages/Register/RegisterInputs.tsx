import dynamic from 'next/dynamic';
import React, { FunctionComponent } from 'react';
import InputContainer from '../../common/Form/InputContainer';
import FirstSection from './Sectionts/FirstSection';
import { Section } from './services/definitions';
const SecondSection = dynamic(() => import('./Sectionts/SecondSection'), {
  ssr: false,
});

interface RegisterInputsProps {
  section: Section;
}

const RegisterInputs: FunctionComponent<RegisterInputsProps> = ({
  section,
}) => {
  return (
    <InputContainer>
      {section === Section.firstSection && <FirstSection />}
      {section === Section.secondSection && <SecondSection />}
    </InputContainer>
  );
};

export default RegisterInputs;
