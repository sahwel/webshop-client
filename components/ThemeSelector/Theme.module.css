
@tailwind components;

@layer components {
    .theme {
       @apply rounded-full p-2  flex justify-center items-center
    }
}