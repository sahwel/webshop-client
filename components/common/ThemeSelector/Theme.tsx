import { FunctionComponent, ReactNode } from 'react';
import cl from 'classnames';
import styles from './theme.module.css';

interface ThemeProps {
  children?: ReactNode;
  isHighlighted: boolean;
  highlightedClass: string;
}

const Theme: FunctionComponent<ThemeProps> = ({
  children,
  isHighlighted,
  highlightedClass,
}) => {
  return (
    <div className={cl({ [highlightedClass]: isHighlighted }, styles.theme)}>
      {children}
    </div>
  );
};

export default Theme;
