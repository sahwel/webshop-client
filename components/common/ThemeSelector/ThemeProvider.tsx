import {
  FunctionComponent,
  ReactNode,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';
import { ThemeContext, ThemeContextModel } from './ThemeContext';
import { Themes } from './definitions';
import Loading from '../Loading/Loading';

interface ThemeProviderProps {
  children?: ReactNode;
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({ children }) => {
  const [currentTheme, setCurrentTheme] = useState<Themes>('light');
  const [isLoaded, setIsLoaded] = useState(false);
  useEffect(() => {
    // todo: get alternative solution
    const theme = localStorage.theme ? localStorage.theme : 'light';

    if (theme === 'dark') {
      document.documentElement.classList.add('dark');
    } else {
      document.documentElement.classList.remove('dark');
    }
    setCurrentTheme(theme);
    setIsLoaded(true);
    return () => {};
  }, []);
  const contextValue: ThemeContextModel = useMemo(
    () => ({
      setTheme() {
        setCurrentTheme((oldTheme) => {
          const newTheme = oldTheme === 'light' ? 'dark' : 'light';

          if (newTheme === 'dark') {
            document.documentElement.classList.add('dark');
          } else {
            document.documentElement.classList.remove('dark');
          }
          localStorage.theme = newTheme;
          return newTheme;
        });
      },
      getTheme() {
        return currentTheme;
      },
    }),
    [currentTheme]
  );
  if (!isLoaded) return null;
  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

export default ThemeProvider;

export function useTheme() {
  const value = useContext(ThemeContext);
  return value;
}
