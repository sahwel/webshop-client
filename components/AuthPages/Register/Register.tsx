import { useCallback, useState } from 'react';
import RegisterFooter from './RegisterFooter';
import RegisterHeader from './RegisterHeader';
import RegisterInputs from './RegisterInputs';
import { Section } from './services/definitions';

const Register = () => {
  const [section, setSection] = useState<Section>(Section.firstSection);
  const handleSectionChange = useCallback((toFoward: boolean) => {
    console.log('asd');
    setSection((oldSection) => oldSection + (toFoward ? 1 : -1));
  }, []);

  return (
    <div>
      <form className="pt-20 grid gap-12 ">
        <RegisterHeader />
        <RegisterInputs section={section} />
        <RegisterFooter
          handleSectionChange={handleSectionChange}
          section={section}
        />
      </form>
    </div>
  );
};

export default Register;
