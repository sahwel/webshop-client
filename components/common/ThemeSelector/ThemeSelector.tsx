import { SunIcon } from '@heroicons/react/outline';
import { MoonIcon } from '@heroicons/react/solid';
import { useCallback, useLayoutEffect, useState } from 'react';
import cl from 'classnames';
import Theme from './Theme';
import styles from './ThemeSelector.module.css';
import { useTheme } from './ThemeProvider';

const ThemeSelector = () => {
  const siteTheme = useTheme();
  const handleChange = useCallback(() => {
    siteTheme.setTheme();
  }, []);

  return (
    <div className={styles.theme_sececltor} onClick={handleChange}>
      <Theme
        highlightedClass={styles.theme_sun_highlited}
        isHighlighted={siteTheme.getTheme() === 'light'}
      >
        <SunIcon className={cl('icon-6', styles.theme_sun)} />
      </Theme>

      <Theme
        highlightedClass={styles.theme_moon_highlited}
        isHighlighted={siteTheme.getTheme() === 'dark'}
      >
        <MoonIcon
          className={cl('icon-6', styles.theme_moon, {
            ['!text-white ']: siteTheme.getTheme() === 'dark',
          })}
        />
      </Theme>
    </div>
  );
};

export default ThemeSelector;
