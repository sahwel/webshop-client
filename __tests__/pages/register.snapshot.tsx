import { render, renderHook } from '@testing-library/react';
import {
  mockUseThemeSelector,
  themeSelectorWrapper,
} from '../../testUtils/MockThemeSelector';
import { useTheme } from '@/components/common/ThemeSelector/ThemeProvider';
import React from 'react';
import Register from '@/pages/register';

React.useContext = mockUseThemeSelector;

describe('register snapshot', () => {
  let _result;
  beforeEach(() => {
    const { result } = renderHook(() => useTheme(), {
      wrapper: themeSelectorWrapper,
    });

    _result = result;
  });
  it('renders homepage unchanged', () => {
    const { container } = render(<Register />);
    expect(container).toMatchSnapshot();
  });
});
