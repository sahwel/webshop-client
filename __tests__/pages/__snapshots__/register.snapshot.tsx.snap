// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register snapshot renders homepage unchanged 1`] = `
<div>
  <div
    class="theme_sececltor"
  >
    <div
      class="theme_sun_highlited theme"
    >
      <svg
        aria-hidden="true"
        class="icon-6 theme_sun"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
    <div
      class="theme"
    >
      <svg
        aria-hidden="true"
        class="icon-6 theme_moon"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
        />
      </svg>
    </div>
  </div>
  <div
    class="auth_pages_container"
  >
    <div
      class="blob_container"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="blob"
          class="blob"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </span>
    </div>
    <form
      class="pt-20 grid gap-12 "
    >
      <div
        class="select-none"
      >
        <h1
          class="text-[300%]"
        >
          Let’s sign up!
        </h1>
        <h3>
          We are happy you’re
        </h3>
        <h3>
           joining us!
        </h3>
      </div>
      <div
        class="input_container"
      >
        <p>
          Please enter some of your personal datas, then hit next!
        </p>
        <div
          class="input_container"
        >
          <input
            class="input"
            placeholder="Email"
            type="text"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-envelope input_icon icon-7"
            data-icon="envelope"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 128C0 92.65 28.65 64 64 64H448C483.3 64 512 92.65 512 128V384C512 419.3 483.3 448 448 448H64C28.65 448 0 419.3 0 384V128zM48 128V150.1L220.5 291.7C241.1 308.7 270.9 308.7 291.5 291.7L464 150.1V127.1C464 119.2 456.8 111.1 448 111.1H64C55.16 111.1 48 119.2 48 127.1L48 128zM48 212.2V384C48 392.8 55.16 400 64 400H448C456.8 400 464 392.8 464 384V212.2L322 328.8C283.6 360.3 228.4 360.3 189.1 328.8L48 212.2z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="input_container"
        >
          <input
            class="input"
            placeholder="First name"
            type="text"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-face-grin input_icon icon-7"
            data-icon="face-grin"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M349.5 308.4C368.2 303.1 385.4 320.4 374.1 336.5C350.4 374.6 306.3 399.1 255.9 399.1C205.6 399.1 161.5 374.6 136.9 336.5C126.5 320.4 143.7 303.1 162.3 308.4C191.3 315.1 222.8 318.8 255.9 318.8C289 318.8 320.6 315.1 349.5 308.4zM208.4 208C208.4 225.7 194 240 176.4 240C158.7 240 144.4 225.7 144.4 208C144.4 190.3 158.7 176 176.4 176C194 176 208.4 190.3 208.4 208zM304.4 208C304.4 190.3 318.7 176 336.4 176C354 176 368.4 190.3 368.4 208C368.4 225.7 354 240 336.4 240C318.7 240 304.4 225.7 304.4 208zM512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256zM256 48C141.1 48 48 141.1 48 256C48 370.9 141.1 464 256 464C370.9 464 464 370.9 464 256C464 141.1 370.9 48 256 48z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="input_container"
        >
          <input
            class="input"
            placeholder="Last name"
            type="text"
          />
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-face-grin input_icon icon-7"
            data-icon="face-grin"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M349.5 308.4C368.2 303.1 385.4 320.4 374.1 336.5C350.4 374.6 306.3 399.1 255.9 399.1C205.6 399.1 161.5 374.6 136.9 336.5C126.5 320.4 143.7 303.1 162.3 308.4C191.3 315.1 222.8 318.8 255.9 318.8C289 318.8 320.6 315.1 349.5 308.4zM208.4 208C208.4 225.7 194 240 176.4 240C158.7 240 144.4 225.7 144.4 208C144.4 190.3 158.7 176 176.4 176C194 176 208.4 190.3 208.4 208zM304.4 208C304.4 190.3 318.7 176 336.4 176C354 176 368.4 190.3 368.4 208C368.4 225.7 354 240 336.4 240C318.7 240 304.4 225.7 304.4 208zM512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256zM256 48C141.1 48 48 141.1 48 256C48 370.9 141.1 464 256 464C370.9 464 464 370.9 464 256C464 141.1 370.9 48 256 48z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
      <div>
        <button
          class="button"
          type="button"
        >
          Next
        </button>
        <a
          class="text-center block text-sm mt-2 cursor-pointer"
          href="/login"
        >
          Back
        </a>
      </div>
    </form>
  </div>
</div>
`;
